src/main/java/net/spy/memcached/ConnectionFactory.java

```diff
@@ -117,7 +117,12 @@ MemcachedNode createMemcachedNode(SocketAddress sa, SocketChannel c,
    * @return the clientMode.
    */
   ClientMode getClientMode();
-  
+
+  /**
+   * Set the client mode in client.
```

Comment:
Also add a comment that it should only be used in the constructor when we discover the client mode.